sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -402,9 +402,78 @@ export class StreamingReceiver extends MessageReceiver {
     await this._receiverHelper.suspend();
   }
 
-  async init(useNewName: boolean, abortSignal?: AbortSignalLike): Promise<void> {
-    const options = this._createReceiverOptions(useNewName, this._getHandlers());
-    await this._init(options, abortSignal);
+  private static wrapRetryOperation(
+    entityPath: string,
+    fullyQualifiedNamespace: string,
+    fn: () => Promise<void>,
+    onError?: OnError
+  ) {
+    return async () => {
+      try {
+        await fn();
+      } catch (err) {
+        onError &&
+          onError({
+            error: err,
+            errorSource: "receive",
+            entityPath,
+            fullyQualifiedNamespace
+          });
+
+        if (err.name !== "AbortError") {
+          err.retryable = true;
```

Comment:
Are we sure we want to update the same error rather than create a new error object and throw that? Since we pass the error to user's error callback, they might be hanging on to it and changing it on them might not be a good idea.

Reply:
Thinking about this more I don't even need to rethrow their original error - it's already been reported so I can just throw _anything_ at this point, so long as it says retryable.

Reply:
Perhaps I'll just do a shallow copy however. Then I can only call onError at the end of the retry() cycle.

Reply:
This is the part which gets me thinking if we should even re-use the `retry()` method. We already have a while loop in place. Why not place a sleep between each iteration of the while loop as long as there is an error and be done with it?

Reply:
Hopefully this was prior to our discussion in teams earlier. :)

Just in case though, I still think this is a valid approach in that it gives users control over the general retries with only one exception (this in between, where we restart the retry cycle).

I created another error type (StreamingReceiverError) that I use to wrap the error and that is always set to retryable. It gets used most of the time and is just an envelope with a permanent 'retryable' flag set on it.

Reply:
Ok, but then we should be treating non retryable errors as retryable. The outer while loop will take care of the retries anyway